Updates for clean-up (#942)
MicrosoftDocs/biztalk-docs: his/core/as-400-appc-printing1.md

---

> description: "Learn more about: IBM System i (APPC) Printing"

Reviewer: ```suggestion
description: Learn more about printing for IBM System i (APPC).
```

> title: "AS-400 (APPC) Printing"

Reviewer: ```suggestion
title: Printing for IBM System i (APPC)
```

ms.custom: ""

> ms.date: "11/30/2017"

Reviewer: ```suggestion
ms.date: 11/28/2023
```

> ms.suite: ""

Reviewer: ```suggestion
```

> ms.topic: "article"

Reviewer: ```suggestion
ms.topic: conceptual
```

---

> # IBM System i (APPC) Printing

Reviewer: ```suggestion
# APPC printing for IBM System i 
```

> APPC printing, like LU 1 printing, uses SCS control codes in the data stream. The set of SCS control codes available for use in APPC printing is more extensive, and allows more formatting options than the set described for LU 1 printing. The IBM IBM System i also provides an additional method for formatting print jobs called Host Print Transform (HPT). With HPT enabled, the IBM System i takes responsibility for rendering the print job into data the printer can understand.  

Reviewer: ```suggestion
APPC printing, like LU 1 printing, uses SCS control codes in the data stream. The set of SCS control codes available for use in APPC printing is more extensive and allows more formatting options than the set described for LU 1 printing. The IBM System i also provides an additional method for formatting print jobs called Host Print Transform (HPT). With HPT enabled, the IBM System i takes responsibility for rendering the print job into data the printer can understand.  
```

>  ![Image that shows the APPC connection between IBM System i and Host Integration Server to send print job to local printer.](../core/media/prn04.gif "prn04")  

Reviewer: Needed a trailing blank line: 

```suggestion
![Image that shows the APPC connection between IBM System i and Host Integration Server to send print job to local printer.](../core/media/prn04.gif)

```

> APPC connection between IBM System i and Host Integration Server to send print job to local printer  

Reviewer: This incomplete sentence looks like a caption for the image? If yes, doesn't belong after the image, but should appear preceding the image and reworded like this:

`The following diagram shows the APPC connection between IBM System i and Host Integration Server for sending a print job to the local printer:`

  
## Host Print Transform (HPT)  

>  When SCS control codes are used by the host to format the print output, a print emulator is responsible for translating the SCS codes and characters into data that the printer can understand, through the Windows printer driver and Windows Print system. With HPT enabled, the IBM System i converts the data to printer control codes before sending the data to Host Integration Server. This output from the host requires no further processing after it leaves the IBM System i. The print emulator's only responsibility is submitting the data to the printer.  

Reviewer: ```suggestion

When SCS control codes are used by the host to format the print output, a print emulator is responsible for translating the SCS codes and characters into data that the printer can understand, through the Windows printer driver and Windows Print system. With HPT enabled, the IBM System i converts the data to printer control codes before sending the data to Host Integration Server. This output from the host requires no further processing after it leaves the IBM System i. The print emulator's only responsibility is submitting the data to the printer.  
```

>  HPT is enabled on the IBM System i in the Device description for the print session. When HPT is enabled, pre-rendered print jobs are sent to the Host Integration Server in marked ASCII Transparent (ATRN) sections using the SCS control code '0x03.' The ATRN control code provides the same function as the Transparent (TRN) control code detailed in the LU 1 printing section. In addition to indicating that the block of data that should be dealt with as transparent, ATRN also indicates that the data is ASCII; therefore it is not converted from EBCDIC to ASCII.  

Reviewer: ```suggestion
HPT is enabled on the IBM System i in the Device description for the print session. When HPT is enabled, pre-rendered print jobs are sent to the Host Integration Server in marked ASCII Transparent (ATRN) sections using the SCS control code '0x03'. The ATRN control code provides the same function as the Transparent (TRN) control code that's detailed in the LU 1 printing section. ATRN indicates that the data block is ASCII, should be handled as transparent, and isn't converted from EBCDIC to ASCII.
```
